File: InterviewPrep/easy/1572_matrixDiagSum.go
/**
 *      file    :  1572_matrixDiagSum.go
 *      author  :  techiedheeraj
 *      created :  2020 Nov 24 01:53:05
 *      lastMod :  Tue Nov 24 01:53:05 2020
**/

func diagonalSum(mat [][]int) int {

  sum:=0
  i:=0
  for i < len(mat[0]) {
    sum+= mat[i][i]
    i++
  }
  k := len(mat)-1
  i=0
  for i < len(mat[0]) {
    if (i == k) {
      i++
      k--
      continue
    }
    sum+=mat[i][k]
    i++
    k--
  }

  return sum
}
 /*
  sum := 0
  for i, _:= range mat {
    for k, _:= range mat[i] {
      if i == k {
        sum+=mat[i][k]
      }
      if i+k == len(mat)-1 && i != k {
        sum+=mat[i][len(mat)-1-i]
      }
    }
  }
  return sum
}
*/

File: InterviewPrep/easy/1572_matrixDiagSum.py
###
 #      file    :  1572_matrixDiagSum.py
 #      author  :  techiedheeraj
 #      created :  2020 Nov 24 01:53:22
 #      lastMod :  Tue Nov 24 01:53:22 2020
###

class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
      summ = 0

      for i in range(len(mat)):
        for j in range(len(mat[0])):
          if (i == j):
            summ+=mat[i][j]
          if i+j == len(mat[0])-1:
            if i == j:
              continue
            summ+=mat[i][len(mat[0])-1-i]
      return summ
